tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -132,6 +132,20 @@ internal static DynamicMethod CreateBeginMethodDelegate(Type integrationType, Ty
                         callGenericTypes.Add(parameterProxyType);
                     }
                 }
+                else if (targetParameterType.IsByRef && targetParameterType.GetElementType().IsGenericParameter)
+                {
+                    // ByRef generic parameters needs to be unwrapped before accessing the `IsGenericParameter` property.
+                    var genTargetParameterType = genericArgumentsTypes[targetParameterType.GetElementType().GenericParameterPosition];
```

Comment:
Not sure if you prefer it, but can avoid calling `GetElementType()` twice if you like?
```suggestion
                else if (targetParameterType.IsByRef && targetParameterType.GetElementType() is { IsGenericParameter: true } elementType)
                {
                    // ByRef generic parameters needs to be unwrapped before accessing the `IsGenericParameter` property.
                    var genTargetParameterType = genericArgumentsTypes[elementType.GenericParameterPosition];
```

Reply:
🤯